remote/remote.go

```diff
@@ -117,16 +138,19 @@ func NewImage(repoName string, keychain authn.Keychain, ops ...ImageOption) (*Im
 		repoName:            repoName,
 		image:               image,
 		addEmptyLayerOnSave: imageOpts.addEmptyLayerOnSave,
+		registries:          imageOpts.registries,
 	}
 
 	if imageOpts.prevImageRepoName != "" {
-		if err := processPreviousImageOption(ri, imageOpts.prevImageRepoName, platform); err != nil {
+		reg := getRegistry(imageOpts.prevImageRepoName, ri.registries)
```

Comment:
It might read better if this function call were inside `processPreviousImageOption`.